### ISSUE TITLE
AdamOptimizer.UpdateSolution: Vector lengths must match (gradient/parameter mismatch) at multi-layer Transformer L=8

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1245]
## Summary

`AiDotNet 0.176.0` (with `AiDotNet.Tensors 0.70.1`): `AdamOptimizer<T>.UpdateSolution` throws `System.ArgumentException : Vector lengths must match. Got 431360 and 524288` when training a `TransformerArchitecture<float>` with `numEncoderLayers=8` via `AiModelBuilder.BuildAsync`. Crash happens during the optimizer's first parameter update step, after forward + backward pass complete successfully.

## Stack trace

```
System.ArgumentException : Vector lengths must match. Got 431360 and 524288
   at AiDotNet.Tensors.Engines.CpuEngine.Add[T](Vector`1 a, Vector`1 b)
   at AiDotNet.Tensors.Engines.DirectGpuTensorEngine.AiDotNet.Tensors.Engines.IEngine.Add[T](Vector`1 a, Vector`1 b)
   at AiDotNet.Optimizers.AdamOptimizer`3.UpdateSolution(IFullModel`3 currentSolution, Vector`1 gradient)
   at AiDotNet.Optimizers.AdamOptimizer`3.Optimize(OptimizationInputData`3 inputData)
   at AiDotNet.AiModelBuilder`3.BuildSupervisedInternalAsync(TInput x, TOutput y)
   at AiDotNet.AiModelBuilder`3.BuildAsync(CancellationToken cancellationToken)
```

## Repro

Build a `TransformerArchitecture<float>` with `numEncoderLayers=8`, dModel=64, dFf=256, numHeads=4, then optimize with `AdamOptimizer`:

```csharp
var arch = new TransformerArchitecture<float>(
    inputType: NeuralNetworkInputType.OneDimensional,
    taskType: NeuralNetworkTaskType.MultiClassClassification,
    inputDim: 64,
    outputDim: 256,
    dModel: 64,
    dFf: 256,
    numHeads: 4,
    numEncoderLayers: 8,
    numDecoderLayers: 0);

var builder = new AiModelBuilder<Matrix<float>, Vector<float>, MeanMetric>()
    .WithArchitecture(arch)
    .WithOptimizer(new AdamOptimizer<Matrix<float>, Vector<float>, MeanMetric>());

await builder.BuildAsync();  // <-- crashes
```

L=1, L=2, L=4 do NOT crash but converge to flat softmax (top-1 = 0%, PPL = V exactly — see related issue #1232 below). L=8 crashes outright.

## Numbers

- Gradient vector length: 431360
- Model parameter vector length: 524288
- Difference: 92928 = exactly 2 · 64 · 64 · 11.34 (could be related to a missing weight matrix in the gradient assembly)

## Impact on HRE program

This blocks the L4 self-attention head-to-head depth sweep at L=8 (PR #105 §5 in HarmonicEngine). The HRE side ran successfully and produced its full L=1/2/4/8 curve; the classical side crashed at L=8 before producing a number.

## Related

- Companion to ooples/AiDotNet#1232 (Transformer flat-softmax convergence, NOT actually fixed in 0.176.0 despite changelog — see below).
- Companion to ooples/AiDotNet#1234 (SequenceTokenSliceLayer deserialization, fixed in 0.174.0+).

## Note on #1232 status

#1232 was reported as fixed in 0.174.0+ but in our 0.176.0 head-to-head test, the Transformer at L=1/2/4 still converges to flat softmax (top-1 = 0%, PPL = 256 = V exactly):

```
L=1: top1=0.00% PPL=256.47 params=82,816 train=24.7s
L=2: top1=0.00% PPL=256.00 params=132,608 train=38.7s
L=4: top1=0.80% PPL=254.70 params=232,192 train=75.5s
```

This is the same pattern #1232 originally reported. Need to re-verify whether the fix was applied to all configurations or just specific ones. If still broken, please reopen #1232 with our updated repro (TransformerArchitecture + AdamOptimizer + byte-LM at d=64, 100KB Shakespeare, 3 epochs).

## Environment

- AiDotNet 0.176.0
- AiDotNet.Tensors 0.70.1
- AiDotNet.Native.OpenBLAS 0.70.1
- .NET 10.0
- Windows 11 Pro 10.0.26220
